healthcare/healthcare/doctype/patient_history_settings/patient_history_settings.py

```diff
@@ -79,31 +81,47 @@ def create_medical_record(doc, method=None):
 	if not medical_record_required:
 		return
 
-	if frappe.db.exists("Patient Medical Record", {"reference_name": doc.name}):
+	reference = doc.name
+	if doc.doctype == "Observation":
+		if doc.reference_docname or doc.sales_invoice:
+			ref_docname = doc.reference_docname
+			if doc.sales_invoice:
+				ref_docname = doc.sales_invoice
+
+			reference = frappe.db.exists("Diagnostic Report", {"docname": ref_docname})
+
+	if frappe.db.exists("Patient Medical Record", {"reference_name": reference}):
+		if doc.doctype == "Observation" and reference:
+			update_medical_record(doc, reference=reference)
 		return
 
-	subject = set_subject_field(doc)
+	subject = set_subject_field(doc, reference)
 	date_field = get_date_field(doc.doctype)
 	medical_record = frappe.new_doc("Patient Medical Record")
 	medical_record.patient = doc.patient
 	medical_record.subject = subject
 	medical_record.status = "Open"
 	medical_record.communication_date = doc.get(date_field)
-	medical_record.reference_doctype = doc.doctype
-	medical_record.reference_name = doc.name
+	medical_record.reference_doctype = (
+		doc.doctype if doc.doctype != "Observation" else "Diagnostic Report"
```

Comment:
how come Diagnostic Report is added to Patient Medical Record?

Reply:
its by getting the related diagnostic report and render diagnostic_report.html. Added patches for creating existing medical records for observations and updated the subject fields

Reply:
let's add medical records corresponding to each Observation (and children if has_components) rather than using the diagnostic report. please note that Observations can be of different category, i.e., Vital Signs, Procedure etc. not just Laboratory